### PR TITLE
[11.0][MIG] mail_environment

### PR DESCRIPTION
**Pull request by simahawk**
replaces https://github.com/OCA/server-tools/pull/1074

### CODE REVIEW

File: __manifest__.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Copyright 2012-2017 Camptocamp SA
```
Comment (ghost) [inline __manifest__.py]: 2018

File: __manifest__.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+{
+    'name': 'Mail configuration with server_environment',
+    'version': '11.0.1.0.0',
+    'category': 'Tools',
+    'summary': 'Configure mail servers with server_environment_files',
+    'author': "Camptocamp,Odoo Community Association (OCA)",
+    'license': 'AGPL-3',
+    'website': 'http://odoo-community.org',
+    'depends': ['fetchmail',
```
Comment (ghost) [inline __manifest__.py]:
```
'depends': [
    'fetchmail',
    ...
],
```

File: __manifest__.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+                'server_environment',
+                ],
+    'data': ['views/fetchmail_server_views.xml',
```
Comment (ghost) [inline __manifest__.py]: same

File: models/__init__.py [hunk 4]
```diff
@@ -0,0 +1,3 @@
+# -*- coding: utf-8 -*-
```
Comment (ghost) [inline models/__init__.py]: can be removed

File: models/fetchmail_server.py [hunk 5]
```diff
@@ -0,0 +1,72 @@
+# Copyright 2012-2017 Camptocamp SA
```
Comment (ghost) [inline models/fetchmail_server.py]: 2018

**Review by ghost — COMMENTED** [review]
LGTM
several minor changes required

**Comment by simahawk** [comment]
damn, I screwed the module path in this porting from the other branch (it ended up in the root)... fix ongoing

File: setup/auditlog/odoo/addons/auditlog [hunk 6]
```diff
@@ -0,0 +1 @@
+../../../../auditlog
```
Comment (ghost) [inline setup/auditlog/odoo/addons/auditlog]: Is this file should be present in commit?

Comment (simahawk) [reply setup/auditlog/odoo/addons/auditlog]: ouch, it came with "[ADD] setup.py". It won't hurt as `setup/` are update by the bot (the same that introduced that commit). I'm gonna drop it anyway :wink:

File: mail_environment/README.rst [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+In the configuration file of each environment, you may first use the
+sections `[outgoing_mail]` and `[incoming_mail]` to configure the
+default values respectively for SMTP servers and the IMAP/POP servers.
+
+Then for each server, you can define additional values or override the
+default values with a section named `[outgoing_mail.resource_name]` or
+`[incoming_mail.resource_name]` where "resource_name" is the name of
+the server.
+
+Exemple of config file ::
```
Comment (ghost) [inline mail_environment/README.rst]: s/Exemple/Example

File: mail_environment/README.rst [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+* Due to the special nature of this addon, you cannot test it on the OCA
+  runbot.
+
+Bug Tracker
+===========
+
+Bugs are tracked on `GitHub Issues
+<https://github.com/OCA/server-tools/issues>`_. In case of trouble, please
+check there if your issue has already been reported. If you spotted it first,
+help us smashing it by providing a detailed and welcomed feedback.
```
Comment (ghost) [inline mail_environment/README.rst]: From [Migration to version 11.0 guidelines](https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-11.0#tasks-to-do-in-the-migration)/[README.rst template](https://raw.githubusercontent.com/OCA/maintainer-tools/master/template/module/README.rst):
 `help us smash it by providing detailed and welcomed feedback.`

File: mail_environment/README.rst [hunk 9]
```diff
@@ -0,0 +1,117 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
```
Comment (ghost) [inline mail_environment/README.rst:1]: use .PNG image:
`.. image:: https://img.shields.io/badge/license-AGPL--3-blue.png`

File: mail_environment/__manifest__.py [hunk 10]
```diff
@@ -0,0 +1,19 @@
+# Copyright 2012-2018 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+{
+    'name': 'Mail configuration with server_environment',
+    'version': '11.0.1.0.0',
+    'category': 'Tools',
+    'summary': 'Configure mail servers with server_environment_files',
+    'author': "Camptocamp,Odoo Community Association (OCA)",
```
Comment (ghost) [inline mail_environment/__manifest__.py]: Shouldn't there be a space after the comma?

File: mail_environment/views/fetchmail_server_views.xml [hunk 11]
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+  <record model="ir.ui.view" id="inherit_fetchmail">
```
Comment (ghost) [inline mail_environment/views/fetchmail_server_views.xml]: A record's ID should be placed before the model, according to [this](https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#security-view-and-action)

**Review by ghost — CHANGES_REQUESTED** [review]

File: mail_environment/README.rst (hunk 9 above)
Comment (simahawk) [reply mail_environment/README.rst:1]: svg is better for 2 reasons: it's lighter and size can be adapted ;)

**Comment by simahawk** [comment]
@arkostyuk all comments addressed

**Review by ghost — APPROVED** [review]

**Review by TDu — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Review by leemannd — APPROVED** [review]

**Review by guewen — APPROVED** [review]

**Pull request merged**